### PR TITLE
Drop PHP 7.3, General Maintenance

### PR DESCRIPTION
**Pull request by gsteel**
|    Q          |   A
|-------------- | ------
| Documentation | yes
| QA            | yes

### Description

- Drops support for 7.3
- Tidy up dotfiles and tool config
- Fix newly discovered psalm errors and update baseline
- Adds `strict_types` everywhere

Closes #23 
Closes #24

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -18,27 +18,28 @@
...
+            "php": "7.4.99"
         },
         "allow-plugins": {
             "dealerdirect/phpcodesniffer-composer-installer": true
         }
     },
     "extra": {},
     "require": {
-        "php": "^7.3 || ~8.0.0 || ~8.1.0",
-        "laminas/laminas-math": "^2.7 || ^3.0",
```
Comment (gsteel) [inline composer.json:30]: Can't see how dropping 2.7 here would be an issue… This lib can't be used with PHP 5 anyhow

File: src/Image.php [hunk 2]
```diff
@@ -597,7 +596,12 @@ protected function generateImage($id, $word)
                               + $colorXY * $fracX * $fracY;
                 }
 
-                imagesetpixel($img2, $x, $y, imagecolorallocate($img2, $newcolor, $newcolor, $newcolor));
+                imagesetpixel($img2, $x, $y, imagecolorallocate(
+                    $img2,
+                    (int) $newcolor,
+                    (int) $newcolor,
+                    (int) $newcolor
+                ));
```
Comment (gsteel) [inline src/Image.php:604]: I think this was the only potential strict_types violation I could find

File: src/ReCaptcha.php [hunk 3]
```diff
@@ -270,11 +271,6 @@ public function isValid($value, $context = null)
             $res = $service->verify($value);
         }
 
-        if (! $res) {
-            $this->error(self::ERR_CAPTCHA);
-            return false;
-        }
-
```
Comment (gsteel) [inline src/ReCaptcha.php:277]: Useless condition

File: composer.json [hunk 4]
```diff
@@ -18,27 +18,28 @@
...
-        "laminas/laminas-recaptcha": "^3.0",
+        "php": "^7.4 || ~8.0.0 || ~8.1.0",
+        "laminas/laminas-math": "^3.5.0",
```
Comment (Ocramius) [inline composer.json]: Still in use?

File: src/AbstractAdapter.php [hunk 5]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+declare(strict_types=1);
```
Comment (Ocramius) [inline src/AbstractAdapter.php]: Let's not do it here

File: composer.json (hunk 4 above)
Comment (gsteel) [reply composer.json]: Just needed `random_int` and `random_bytes` to drop it completely

**Review by Ocramius — APPROVED** [review]
LGTM 👍

**Pull request merged**